Sources/TensorFlow/Layers/Core.swift

```diff
@@ -12,67 +12,19 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-fileprivate extension Tensor where Scalar: TensorFlowFloatingPoint {
-     /// Computes dropout given a probability.
-     // TODO: Remove the underscore once `droppingOut(probability:)` has been removed.
-    @differentiable(wrt: self where Scalar: Differentiable)
-    func _droppingOut(probability: Double) -> Tensor {
-        let noise = Tensor(randomUniform: shape)
-        let keepMask = noise .>= Scalar(probability)
-        let keepProbability = Scalar(1.0 - probability)
-        return self * Tensor(keepMask) / Tensor(keepProbability)
-    }
-}
-
-public extension Tensor where Scalar: TensorFlowFloatingPoint {
-    /// Computes dropout given a probability.
-    @available(*, deprecated, message: """
-        This API will be removed after Swift for TensorFlow 0.6.
-        For dropout, use the `Dropout` layer.
-        """)
-    @differentiable(wrt: self where Scalar: Differentiable)
-    func droppingOut(probability: Double) -> Tensor {
-        _droppingOut(probability: probability)
-    }
-}
-
-/// A dropout layer.
-///
-/// Dropout consists in randomly setting a fraction of input units to `0` at each update during
-/// training time, which helps prevent overfitting.
-@frozen
-public struct Dropout<Scalar: TensorFlowFloatingPoint>: ParameterlessLayer {
-    @noDerivative public let probability: Double
-
-    /// Creates a dropout layer.
-    ///
-    /// - Parameter probability: The drop probability.
-    public init(probability: Double) {
-        self.probability = probability
-    }
+/// A layer that encloses a custom differentiable function.
+public struct Function<Input: Differentiable, Output: Differentiable>: ParameterlessLayer {
```

Comment:
Could you keep the ordering of the layers the same as before? (This will help make the history easier to follow.)